### PR TITLE
Initial pull request

### PR DESCRIPTION
**Pull request by jbathmann**
This is quite a big PR - sorry
The PR contains

- TreeModelLib:
    Module containing tree functionalities as Aboveground- and Belowgroundcompetition and the growth concepts
    For all three submodules, simple test scenarios are created as discussed.
    The argument stucture for the Competitions is as follows: assessing the tree positions, geometry and parameters the concept is able to calculate the resources for each individual
    GrowthAndDeathDynamics are using the calculated resources in order to update tree geometry and parameters
    Order of Function-calls per timestep is given in the TimeLoopLib

- PopulationLib:
   Module which structures the tree population into groups. For each group, a species can be defined. Hereby, we keep the structure flexible. As an example, Bettina configurations are passed to tree parameters and geometry. In general, this framework allows to introduce arbitrary geometric and parametric configurations as long as they agree with the competition concepts used.
    Functionality to randomly distribute trees within a specified domain included

- TimeLoopLib
   This contains the functionalities essential for timestepping in tree dynamics
   TreeDynamicTimeStepping structures the order of successive steps per timestep

-ProjectLib
   The whole structure is controlled using xml files. Example containing the currently implemented tags is given.
   Project.py is the heart of the project, since this class links all essential information.

### CODE REVIEW

File: TimeLoopLib/TreeDynamicTimeLoop.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    def getNextTimeStepBoundaries(self):
+        self.t_ini, self.t_end, self.step_on = self.loop.getNextTimeStep()
+
+    def runTimeLoop(self, project):
+        self.getNextTimeStepBoundaries()
+        while (self.step_on):
+            print("Next time step to propagate" +
+                  " tree population with starting time " + str(self.t_ini) +
+                  " and end time " + str(self.t_end) + ".")
+            self.getNextTimeStepBoundaries()
```
Comment (mcwimm) [inline TimeLoopLib/TreeDynamicTimeLoop.py]: does not progress trees

Comment (jbathmann) [reply TimeLoopLib/TreeDynamicTimeLoop.py]: fixed in commit 21ee6e9

File: ProjectLib/XMLtoProject.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        tree = etree.parse(self.prjfile)
+        self.root = tree.getroot()
+        for tag in self.root.iter():
+            tag.text = tag.text.strip()
+
+    def addTreeDynamicConcepts(self):
+        self.tree_dynamics = self.findChild(self.root, "tree_dynamics")
+        self.prj_arguments["aboveground_competition"] = self.findChild(
+            self.tree_dynamics, "aboveground_competition")
+        self.args["belowground_competition"] = self.findChild(
```
Comment (mcwimm) [inline ProjectLib/XMLtoProject.py]: self.prj_arguments instead of self. args (again in line 38)

Comment (jbathmann) [reply ProjectLib/XMLtoProject.py]: fixed

**Comment by jbathmann** [comment]
asdf

File: PopulationLib/GroupPlanting.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
```
Comment (mcwimm) [inline PopulationLib/GroupPlanting.py]: Looks ok!

**Pull request merged**